### ISSUE TITLE
Room upgrade does not sync room members

### ISSUE THREAD

**Issue by vranki** [matrix-org/matrix-appservice-irc#1260]
**Describe the bug**

Upgrading bridged room does not join IRC users from old room to new.

Users join only after when they send their first message in the new room.

Not sure if this happens every time.

**To Reproduce**

1. Go to IRC Bridged channel
2. /upgraderoom 6
3. Join the new room
4. Observe room has no members from IRC until they say something in the channel.

**Expected behavior**

< Half-Shot> Room upgrades should sync users across

**Additional context**

IRCNet, should be running up to date bridge.

Possibly related: #506 #1164

**Comment by vranki** [comment]
I've understood after this that lazy joining is the desired behavior to prevent join floods in large rooms. If this is correct, this issue can be closed. Of course it would be nice to have a command to request full sync if needed and/or maybe sync "small" (<100 users?) rooms fully after upgrade.
